### DISCUSSION TITLE
使用Consul作为服务发现时，无法正确获取upstream host和address

### DISCUSSION THREAD

**Discussion by yw-mao** [higress-group/higress#831] [Q&A]
环境：Docker Compose
版本：1.3.4
Console 版本：1.3.1

使用外部域名时正常解析
`"upstream_cluster":"outbound|7001||ebl.dns","upstream_host":"172.22.0.23:7001","upstream_local_address":"172.22.0.7:40070","upstream_service_time":"7"`

使用Consul时
```
"upstream_cluster":"outbound|7001||xxxxx.dc1.consul","upstream_host":"-","upstream_local_address":"-","upstream_service_time":"-","upstream_transport_failure_reason":"-",
```

**Comment by CH3CHO** [comment]
请提供具体的服务来源和路由配置，以及 Consul 上服务的注册信息。

**Reply by yw-mao** [reply]
服务来源
![image](https://github.com/alibaba/higress/assets/88931049/5493a97f-fd56-4107-aeb3-987e39cd1860)

服务列表已经能够正常识别出后端服务IP
![image](https://github.com/alibaba/higress/assets/88931049/b1e1b23b-8903-4422-9ccc-7e7fbf2dbf8c)

路由是最简单的
![image](https://github.com/alibaba/higress/assets/88931049/91b7e2a4-5fa0-4efc-b8be-78dcf2bf6547)

![image](https://github.com/alibaba/higress/assets/88931049/0c0a84df-106e-4b5d-8da8-f5423d4e6ba1)

**Comment by pianzhou2021** [comment]
这么久了，问题依旧，找了半天资料，发现还是一样的问题

---
After so long, the problem remains the same. After searching for information for a long time, I found that the same problem is still
